Detectors/TPC/simulation/src/DigitGlobalPad.cxx

```diff
@@ -27,16 +23,17 @@
 
 using namespace o2::TPC;
 
-void DigitPad::fillOutputContainer(std::vector<o2::TPC::Digit> *output, o2::dataformats::MCTruthContainer<o2::MCCompLabel> &mcTruth,
-                                   std::vector<o2::TPC::DigitMCMetaData> *debug, int cru, int timeBin, int row, int pad, float commonMode)
+void DigitGlobalPad::fillOutputContainer(std::vector<o2::TPC::Digit> *output, o2::dataformats::MCTruthContainer<o2::MCCompLabel> &mcTruth,
+                                         std::vector<o2::TPC::DigitMCMetaData> *debug, CRU cru, int timeBin, GlobalPadNumber globalPad, float commonMode)
 {
+  const static Mapper &mapper = Mapper::instance();
+  const PadPos pad = mapper.padPos(globalPad);
+
   /// The charge accumulated on that pad is converted into ADC counts, saturation of the SAMPA is applied and a Digit is created in written out
   const float totalADC = mChargePad - commonMode; // common mode is subtracted here in order to properly apply noise, pedestals and saturation of the SAMPA
 
-  float noise = 0.f;
-  float pedestal = 0.f;
-
-  const float mADC = SAMPAProcessing::makeSignal(totalADC, PadSecPos(CRU(cru).sector(), PadPos(row, pad)), pedestal, noise);
+  float noise, pedestal;
+  const float mADC = SAMPAProcessing::makeSignal(totalADC, PadSecPos(CRU(cru).sector(), pad), pedestal, noise);
```

Comment:
I guess since the type is now `CRU` in the function call (should it be `const CRU& cru`?) we can simplify to
`PadSedPos(cru.sector(), pad)`